Sources/MUXSDKStats/MUXSDKPlayerBinding.m

```diff
@@ -440,6 +441,58 @@ - (void)monitorAVPlayerItem {
         
+        // Handle being attached to the monitor late. In this case, we will likely miss all of the
+        //  KVO updates and NSNotificaitons that would set up the metadata we are trying to get here.
+        //  checkVideoData requires (and assumes) that we see those events, but this isn't feasible
+        //  for customers that want to pre-load media or pool their player instances. In those cases
+        //  we need to catch-up some of our state
+        AVPlayerItemTrack *alreadyLoadedVideoTrack = [self findVideoTrackOnPlayerItem:_playerItem];
+        if (alreadyLoadedVideoTrack) {
+            CGSize srcDimens = [self getSourceDimensions];
+            
+            MUXSDKVideoData *videoData = [[MUXSDKVideoData alloc] init];
+            MUXSDKDataEvent *dataEvent = [[MUXSDKDataEvent alloc] init];
+            dataEvent.videoData = videoData;
+            
+            if (srcDimens.width > 0 || srcDimens.height > 0) {
+                // media was already loaded, get the source dimensions so we can catch up
+                _videoSize = srcDimens;
+                _lastDispatchedVideoSize = srcDimens; // At the end of the method, this will reflect reality
+                
+                videoData.videoSourceWidth = [NSNumber numberWithDouble:_videoSize.width];
+                videoData.videoSourceHeight = [NSNumber numberWithDouble:_videoSize.height];
+            }
+            
+            NSString *srcUrlString = [self getURLStringIfAvailableOnAVPlayerItem:_playerItem];
+            if (srcUrlString) {
+                _videoURL = srcUrlString;
+                videoData.videoSourceUrl = srcUrlString;
+            }
+            
+            NSError *error = nil;
+            AVKeyValueStatus status = [_playerItem.asset statusOfValueForKey:@"duration" error: &error];
+            if (!error && status == AVKeyValueStatusLoaded) {
+                CMTime duration = [_playerItem.asset duration];
+                float floatingSec = CMTimeGetSeconds(duration);
+                if (!isnan(floatingSec) && floatingSec > 0) {
+                    float ms = floatingSec * 1000;
+                    NSNumber *timeMs = [NSNumber numberWithFloat:ms];
+                    [videoData setVideoSourceDuration:[NSNumber numberWithLongLong:[timeMs longLongValue]]];
+                }
+                _videoDuration = duration;
+            }
+            
+            long alreadyAdvertisedBitrate = [self findMostRecentAdvertisedBitrateForPlayerItem:_playerItem];
+            if (alreadyAdvertisedBitrate > 0) {
+                _lastAdvertisedBitrate = alreadyAdvertisedBitrate;
+                _lastDispatchedAdvertisedBitrate = alreadyAdvertisedBitrate;
+                videoData.videoSourceAdvertisedBitrate = [
+                    NSNumber numberWithFloat:alreadyAdvertisedBitrate
+                ];
+            }
+            [MUXSDKCore dispatchEvent:dataEvent forPlayer:[self name]];
```

Comment:
It seems like this (and possibly the surrounding code) duplicates what's already in `checkVideoData`, maybe there's a way to avoid that?